netcontrold/tests/lib/test_data_coll.py

```diff
@@ -0,0 +1,287 @@
+status              : {driver_name=net_ixgbe, if_descr="DPDK 18.11.2 net_ixgbe", if_type="6", link_speed="10Gbps", max_hash_mac_addrs="4096", max_mac_addrs="127", max_rx_pktlen="1518", max_rx_queues="128", max_tx_queues="64", max_vfs="0", max_vmdq_pools="64", min_rx_bufsize="1024", numa_id="0", pci-device_id="0x10fb", pci-vendor_id="0x8086", port_no="1"}
+type                : dpdk"""
+
+
+class TestDataif_Collection(TestCase):
+    """
+    Test for getting pmd stats.
+    """
+
+    # create an empty pmd_map
+    pmd_map = dict()
+
+    def setUp(self):
+        # turn off limited info shown in assert failure for pmd object.
+        self.maxDiff = None
+
+        # a noop handler for debug info log.
+        class NlogNoop(object):
+
+            def debug(self, *args):
+                None
+
+        dataif.Context.nlog = NlogNoop()
+
+        # create one pmd object.
+        self.fx_pmd_1 = dataif.Dataif_Pmd(1)
+
+        # let it be in numa 0.
+        self.fx_pmd_1.numa_id = 0
+
+        # add it to pmd_map
+        self.pmd_map[1] = self.fx_pmd_1
+
+        # create port class of name 'virtport1'.
+        dataif.make_dataif_port("port1")
+
+        # add port object into pmd.
+        self.port1 = self.fx_pmd_1.add_port("port1")
+        self.port1.numa_id = self.fx_pmd_1.numa_id
+
+        # add rxq to port object
+        self.rxq_1 = self.port1.add_rxq(0)
+
+        # create another pmd object.
+        self.fx_pmd_2 = dataif.Dataif_Pmd(13)
+
+        # let it be in numa 0.
+        self.fx_pmd_2.numa_id = 0
+
+        # add it to pmd_map
+        self.pmd_map[13] = self.fx_pmd_2
+
+        # create port class of name 'virtport2'.
+        dataif.make_dataif_port("port2")
+
+        # add port object into pmd.
+        self.port2 = self.fx_pmd_2.add_port("port2")
+        self.port2.numa_id = self.fx_pmd_2.numa_id
+
+        # add rxq to port object
+        self.rxq_2 = self.port2.add_rxq(0)
+
+        return
+
+    # Test case:
+    #   getting pmd stats from get_pmd_stats function and checking if
+    #   declared pmd_map is modified or not
+    @mock.patch('netcontrold.lib.util.exec_host_command', mock_pmd_stats)
+    def test_get_pmd_stats_1(self):
+        # create a copy of original pmd_map
+        expected = copy.deepcopy(self.pmd_map)
+
+        # copy both original pmd objects
+        expected_pmd_1 = expected[1]
```

Comment:
Fyi it is not copy here but just another variable pointing to pmd1 in expected.!

Reply:
ya.. i will change the comment